### DISCUSSION TITLE
How does href in Link work?

### DISCUSSION THREAD

**Discussion by Seybo** [vercel/next.js#14545] [Help] [answered]
I have a project structure like this:
![image](https://user-images.githubusercontent.com/10584167/85651097-aec1c080-b6c4-11ea-899d-14736f2b36f1.png)

As you can see my dynamic projects page has name `[id].js` but my code `href="projects/[bar]"` and it works. I can even make it `href="foo"` and it also works. A can click on the link and get to the project's page. 

I was thinking that it automatically looks for the same folder as `as` is pointing to and looks there for any dynamic route page. But then i tried to rename the page folder and set it in `href` like this but it doesn't work.
![image](https://user-images.githubusercontent.com/10584167/85651429-99996180-b6c5-11ea-81be-714b5e4ca71d.png)

And i would expect quite the opposite: the first example to fail and the second to work. So what am i missing here? How does `href` actually work in `Link`?

**Comment by jamesmosier** [comment, marked as answer]
I think what you are seeing when you have an "incorrect" `href` like `href="projects/[bar]"` is that since `next/link` can't find that route it just default to the `as` value and just navigates to that url like any normal `<a href>` would do. If you look at your console you'll see a 404 for the prefetch of `projects/[bar]`. Normally when using next/link with the correct `href` it would do a client side render but in this case it does a full page refresh because Next can't find that page in the pages folder.
